docs/web3.main.rst

```diff
@@ -49,6 +49,85 @@ Attributes
        'Geth/v1.4.11-stable-fed692f6/darwin/go1.7'
 
 
+.. _batch_requests:
+
+Batch Requests
+~~~~~~~~~~~~~~
+
+.. py:method:: Web3.batch_requests()
+
+    The JSON-RPC API allows for batch requests, meaning you can send a single request
+    that contains an array of RPC commands. Generally, this may be useful when you want
+    to limit the number of requests you send to a node.
+
+    You have can choose to build a batch of requests within or outside of a context manager:
+
+    .. code-block:: python
+
+        with w3.batch_requests() as batch:
+            batch.add(w3.eth.get_block(6))
+            batch.add(w3.eth.get_block(4))
+            batch.add(w3.eth.get_block(2))
+
+            responses = batch.execute()
+            assert len(responses) == 3
+
+    Using the batch object directly:
+
+    .. code-block:: python
+
+        batch = w3.batch_requests()
+        batch.add(w3.eth.get_block(1))
+        batch.add(w3.eth.get_block(2))
+        responses = batch.execute()
+        assert len(responses) == 2
+
+    Contract interactions can be included in batch requests by omitting the ``call`` method:
+
+    .. code-block:: python
+
+        batch.add(math_contract.functions.multiply7(0))
+
+    Additionally, if you need to make multiple calls of the same function, you can add
+    a mapping of the function to its arguments:
+
+    .. code-block:: python
+
+        batch = w3.batch_requests()
+        batch.add_mapping(
+            {math_contract.functions.multiply7: [1, 2], w3.eth.get_block: [3, 4]}
```

Comment:
nit: keep or leave, I feel like the formatting can be clearer as more of a block here

```suggestion
            {
                math_contract.functions.multiply7: [1, 2], 
                w3.eth.get_block: [3, 4],
            }
```